src/CreepLaw/PeierlsCreep.jl

```diff
@@ -0,0 +1,318 @@
+    
+    Name = String(collect(p.Name))
+    n = Value(p.n)
+    q = Value(p.q)
+    o = Value(p.o)
+    TauP = uconvert(Pa, Value(p.TauP))
+    A_Pa = uconvert(s^(-1.0), Value(p.A))
+    E_J = uconvert(J / mol, Value(p.E))
+
+    Apparatus = p.Apparatus
+
+    # args from database
+    args = (Name=Name, n=n, q=q, o=o, TauP=TauP, A=A_Pa, E=E_J, Apparatus=Apparatus)
+    
+    return PeierlsCreep(; args...)
+end
+
+"""
+    s = remove_tensor_correction(s::PeierlsCreep)
+
+Removes the tensor correction of the creeplaw, which is useful to compare the implemented creeplaws
+with the curves of the original publications, as those publications usually do not transfer their data to tensor format.
+"""
+function remove_tensor_correction(s::PeierlsCreep)
+    name = String(collect(s.Name))
+
+    return PeierlsCreep(;
+        Name=name, n=s.n, q=s.q, o=s.o, TauP=s.TauP, A=s.A, E=s.E, Apparatus=Invariant
+    )
+end
+
+function param_info(s::PeierlsCreep)
+    name = String(collect(s.Name))
+    eq = L"\tau_{ij} = 2 \eta  \dot{\varepsilon}_{ij}"
+    if name == ""
+        return MaterialParamsInfo(; Equation=eq)
+    end
+    inf = PeierlsCreep_info[name][2]
+    return MaterialParamsInfo(;
+        Equation=eq, Comment=inf.Comment, BibTex_Reference=inf.BibTex_Reference
+    )
+end
+
+# Calculation routines for linear viscous rheologies
+# All inputs must be non-dimensionalized (or converted to consistent units) GeoUnits
+@inline function compute_εII(
+    a::PeierlsCreep, TauII::_T; T=one(precision(a)), args...
+) where {_T}
+    @unpack_val n, q, o, TauP, A, E, R = a
+    FT, FE = a.FT, a.FE
+
+    ε = (A * exp(-(E / (R * T)) * (fastpow(1.0 - fastpow((FT * TauII) / TauP, q), o)))) / FE
+
+    return ε
+end
+
+@inline function compute_εII(
+    a::PeierlsCreep, TauII::Quantity; T=1K, args...
+)
+    @unpack_units n, q, o, TauP, A, E, R = a
+    FT, FE = a.FT, a.FE
+
+    ε = (A * exp(-(E / (R * T)) * (fastpow(1 - fastpow((FT * TauII) / TauP, q), o)))) / FE
+
+    return ε
+end
+
+function compute_εII!(
+    EpsII::AbstractArray{_T,N},
+    a::PeierlsCreep,
+    TauII::AbstractArray{_T,N};
+    T=ones(size(TauII))::AbstractArray{_T,N},
+    kwargs...,
+) where {N,_T}
+    @inbounds for i in eachindex(EpsII)
+        EpsII[i] = compute_εII(a, TauII[i]; T=T[i])
+    end
+
+    return nothing
+end
+
+@inline function dεII_dτII(
+    a::PeierlsCreep, TauII::_T; T=one(precision(a)), args...
+) where {_T}
+    @unpack_val n, q, o, TauP, A, E, R = a
+    FT, FE = a.FT, a.FE
+
+    return q * 
+           fastpow((FT * TauII) / TauP, q) *
+           fastpow(1.0 - fastpow((FT * TauII) / TauP, q), o) *
+           A *
+           E *
+           o *
+           exp(-(E * fastpow(1 - fastpow((FT * TauII) / TauP, q), o)) / (R * T)) /
+           (FE * R * T * TauII)
+end
+
+@inline function dεII_dτII(
+    a::PeierlsCreep, TauII::Quantity; T=1K, args...
+)
+    @unpack_units n, q, o, TauP, A, E, R = a
+    FT, FE = a.FT, a.FE
+
+    return o * 
```

Comment:
Check here as well if ForwardDiff is faster than the symbolic expression.

Reply:
It seems that the symbolic expression is faster:
```julia
# symbolic
@btime dεII_dτII($x1, $TauII, $args)
  62.567 ns (0 allocations: 0 bytes)
3.2013187520023984e-55
 
 # ForwardDiff
 @btime dεII_dτII($x1, $TauII, $args)
  70.255 ns (0 allocations: 0 bytes)
3.2078554209110495e-55
 ```